presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/MetastoreUtil.java

```diff
@@ -444,4 +448,18 @@ else if (type instanceof TinyintType
 
         return withColumnDomains(domains);
     }
+
+    public static Map<String, String> adjustRowCount(Map<String, String> parameters, String description, long rowCountAdjustment)
+    {
+        Map<String, String> parms = new HashMap<>(parameters);
+        String existingRowCount = parameters.get(NUM_ROWS);
+        if (existingRowCount == null) {
+            return parameters;
+        }
+        checkArgument(DIGITS.matcher(existingRowCount).matches(), "For %s, the existing row count (%s) is not a digit string", description, existingRowCount);
+        long newRowCount = Long.parseLong(existingRowCount) + rowCountAdjustment;
+        checkArgument(newRowCount >= 0, "For %s, the subtracted row count (%s) is less than zero, existing count %s, rows deleted %d", description, newRowCount, existingRowCount, rowCountAdjustment);
+        parms.put(NUM_ROWS, String.valueOf(newRowCount));
+        return ImmutableMap.copyOf(parms);
```

Comment:
Unfortunately, using ImmutableMap.Builder doesn't work here, because it doesn't permit duplicate keys, and the putAll() call will add the existing NUM_ROWS.

Your ordering and naming suggestions are good ones, and I've adopted those.